### PR TITLE
Upgrade mysql_common to latest version

### PR DESCRIPTION
**Pull request by mjgarton**
This updates mysql_common to 0.28.0 and adjusts code accordingly.

This is the first of a number of independent PRs to upgrade various dependencies to recent versions.

### CODE REVIEW

**Comment by jonhoo** [comment]
I take it this then effectively replaces #22 (along with your later PRs)?

I'm curious how you ended up working past the round-trip problems that came up last time around when we bumped this in that PR?

Also, it appears the style linting check fails?

**Comment by mjgarton** [comment]
> I take it this then effectively replaces #22 (along with your later PRs)?
> 
> I'm curious how you ended up working past the round-trip problems that came up last time around when we bumped this in that PR?
> 
> Also, it appears the style linting check fails?

I'm not sure it was clear, but #22 is not my PR.   (_my_ only other pending one is #27, but is unrelated to version updates)

In any case, thanks for the review and I will take a look at the linter issues here and push an update.

**Comment by mjgarton** [comment]
The linter issues in here appear unrelated to my changes, so I have fixed them in a separate independent PR  https://github.com/jonhoo/msql-srv/pull/32 .  Once that's merged, I'll merge it into this and my other pending PRs so that each PR should then make the linter happy.

File: src/params.rs
Comment (jonhoo) [inline src/params.rs:76]: We should arguably change `Item` here to `Result<Item>` to propagate errors from here, but can do that as part of a breaking change. In the meantime, would you mind changing this to
```rust
.ok_or_else(|| panic!("got unsupported column type 0x{:x}", typmap[2 * i as usize]))
```
so that the error a user might get is a little more helpful?

File: src/value/decode.rs
Comment (jonhoo) [inline src/value/decode.rs:316]: Do we know that this will serialize using the binary protocol and not the text protocol?

**Review by jonhoo — CHANGES_REQUESTED** [review]
This looks great! It seems like you managed to get around the problems that were blocking #22, which is good news :) Just one nit and one question, and then I think this is ready to land with the lint fix that I just merged (thanks for that!).

File: src/params.rs
Comment (mjgarton) [reply src/params.rs:76]: Done.

File: src/value/decode.rs
Comment (mjgarton) [reply src/value/decode.rs:316]: It seems so.  The implementation of `MySerialize` for `Value` seems to operate always directly in binary as far as I understand it.  See https://github.com/blackbeam/rust_mysql_common/blob/master/src/value/mod.rs#L73

Comment (jonhoo) [reply src/value/decode.rs:312]: Huh, yeah, it seems that way even though it's not documented. I suppose the tests passing is a good indication that this is reliable.

File: src/params.rs
Comment (jonhoo) [inline src/params.rs:79]: Sorry, missed this in my earlier comment — we should also propagate the `|e|` here in case it contains relevant information.

```suggestion
                            .unwrap_or_else(|e| {
                                panic!("bad column type 0x{:x}: {}", typmap[2 * i as usize], e)
                            }),
```

**Review by jonhoo — CHANGES_REQUESTED** [review]

**Review by jonhoo — APPROVED** [review]

**Pull request merged**
